lte/gateway/python/integ_tests/s1aptests/test_attach_service_with_multi_pdns_and_bearers_failure.py

```diff
@@ -0,0 +1,361 @@
+        )
+
+        print("Sleeping for 5 seconds")
+        time.sleep(5)
+        # Verify if flow rules are created
+        dl_flow_rules = {
+            default_ip: [flow_list1],
+            sec_ip: [flow_list2],
+        }
+        # 1 UL flow is created per bearer
+        num_ul_flows = 4
+        # Verify if flow rules are created
+        self._s1ap_wrapper.s1_util.verify_flow_rules(
+            num_ul_flows, dl_flow_rules
+        )
+
+        print("************************* Sleeping for 5 seconds")
+        time.sleep(5)
+        print(
+            "************************* Sending UE context release request ",
+            "for UE id ",
+            ue_id,
+        )
+        # Send UE context release request to move UE to idle mode
+        req = s1ap_types.ueCntxtRelReq_t()
+        req.ue_Id = ue_id
+        req.cause.causeVal = gpp_types.CauseRadioNetwork.USER_INACTIVITY.value
+        self._s1ap_wrapper.s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_CNTXT_REL_REQUEST, req
+        )
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_CTX_REL_IND.value
+        )
+
+        # Send the bearers to be included in failed to setup list in ICS Rsp
+        init_ctxt_setup_failed_erabs = s1ap_types.UeInitCtxtSetupFailedErabs()
+        init_ctxt_setup_failed_erabs.ue_Id = ue_id
+        init_ctxt_setup_failed_erabs.failedErabs[
+            0
+        ] = act_ded_ber_req_oai_apn.bearerId
+        init_ctxt_setup_failed_erabs.failedErabs[
+            1
+        ] = act_ded_ber_req_ims_apn.bearerId
+        init_ctxt_setup_failed_erabs.numFailedErabs = 2
+        self._s1ap_wrapper._s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_SET_INIT_CTXT_SETUP_RSP_FAILED_ERABS,
+            init_ctxt_setup_failed_erabs,
+        )
+
+        # Verify if paging flow rules are created
+        ip_list = [default_ip, sec_ip]
+        self._s1ap_wrapper.s1_util.verify_paging_flow_rules(ip_list)
+
+        print("************************* Sleeping for 5 seconds")
+        time.sleep(5)
+        print(
+            "************************* Sending Service request for UE id ",
+            ue_id,
+        )
+        # Send service request to reconnect UE
+        req = s1ap_types.ueserviceReq_t()
+        req.ue_Id = ue_id
+        req.ueMtmsi = s1ap_types.ueMtmsi_t()
+        req.ueMtmsi.pres = False
+        req.rrcCause = s1ap_types.Rrc_Cause.TFW_MO_DATA.value
+        self._s1ap_wrapper.s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_SERVICE_REQUEST, req
+        )
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.INT_CTX_SETUP_IND.value
+        )
+
+        print("Sleeping for 5 seconds")
+        time.sleep(5)
+        # Verify if flow rules are created
+        dl_flow_rules = {
+            default_ip: [],
+            sec_ip: [],
```

Comment:
don't we want to verify default bearer rules in DL?

Reply:
Whether DL flow is created for the default bearer or not will be verified in s1ap utils function _verify_dl_flow. For default bearer, we do not define any flow rules in the test case, these rules are defined only for dedicated bearers. Hence we dont have to pass from the test case